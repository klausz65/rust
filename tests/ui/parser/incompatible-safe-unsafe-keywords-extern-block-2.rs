//! Check that we don't try to suggest reordering incompatible keywords `safe` and `unsafe` when
//! parsing things that looks like fn frontmatter/extern blocks.
//!
//! # References
//!
//! See <https://github.com/rust-lang/rust/issues/133586>.
//!
//! See `incompatible-safe-unsafe-keywords-extern-block-1.rs` for the `unsafe safqe extern {}`
//! version.
#![crate_type = "lib"]

unsafe safe extern {}
//~^ ERROR expected one of `extern` or `fn`, found `safe`
//~| NOTE expected one of `extern` or `fn`
//~| HELP `unsafe` and `safe` are incompatible, use only one of the keywords
