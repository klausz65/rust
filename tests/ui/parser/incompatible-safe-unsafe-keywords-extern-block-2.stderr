error: expected one of `extern` or `fn`, found `safe`
  --> $DIR/incompatible-safe-unsafe-keywords-extern-block-2.rs:12:8
   |
LL | unsafe safe extern {}
   |        ^^^^ expected one of `extern` or `fn`
   |
help: `unsafe` and `safe` are incompatible, use only one of the keywords
  --> $DIR/incompatible-safe-unsafe-keywords-extern-block-2.rs:12:1
   |
LL | unsafe safe extern {}
   | ^^^^^^^^^^^

error: aborting due to 1 previous error

